### PR TITLE
Logger extracted as dependencies (#396)

### PR DESCRIPTION
**Pull request by mgordel**

### CODE REVIEW

**Comment by Braunmann** [comment]
@mgordel  `utils.changeLogLevel("debug");` seems to be not avaliable anymore. I think it will be nice to have same interface .

File: yajsapi/utils/index.ts [hunk 1]
```diff
@@ -6,13 +6,13 @@ import Callable from "./callable";
 import CancellationToken from "./cancellationToken";
 import eventLoop from "./eventLoop";
 import getAllProperties from "./getAllProperties";
-import logger, * as logUtils from "./log";
-import { changeLogLevel } from "./log";
```
Comment (Braunmann) [inline yajsapi/utils/index.ts:10]: Seems like `utils.changeLogLevel("debug");` is not avaliable anymore - I think it is good idea to do not change interface, or correct examples

Comment (mgordel) [reply yajsapi/utils/index.ts:10]: Yes, you're right. I forgot about examples. Now logger is a param of executor constructor and has method setLevel to change log level. For the default logger (winston for nodejs), when the param is null, we may add logLevel parameter to the executor constructor. I'll fix it.

Comment (mgordel) [reply yajsapi/utils/index.ts:10]: @Braunmann please review again

File: yajsapi/rest/payment.ts [hunk 2]
```diff
@@ -338,9 +338,9 @@ export class Payment {
           "getInvoiceEvents"
         );
         for (const ev of events) {
-          self.logger?.debug(
-            `Received invoice event: ${JSON.stringify(ev)}, ` + `type: ${JSON.stringify(Object.getPrototypeOf(ev))}`
-          );
+          // self.logger?.debug(
```
Comment (Braunmann) [inline yajsapi/rest/payment.ts:341]: Is it consciously commented out?

File: yajsapi/rest/payment.ts [hunk 3]
```diff
@@ -382,9 +382,9 @@ export class Payment {
           "getDebitNoteEvents"
         );
         for (const ev of events) {
-          self.logger?.debug(
-            `Received debit note event: ${JSON.stringify(ev)}, ` + `type: ${JSON.stringify(Object.getPrototypeOf(ev))}`
-          );
+          // self.logger?.debug(
```
Comment (Braunmann) [inline yajsapi/rest/payment.ts:385]: Is it consciously commented out?

Comment (mgordel) [reply yajsapi/rest/payment.ts:385]: Yes, temporary I disabled this because it caused bug described in #381

**Pull request merged**